linalg/dual.hpp

```diff
@@ -0,0 +1,413 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+/**
+ * @file dual.hpp
+ *
+ * @brief This file contains the declaration of a dual number class
+ */
```

Comment:
I don't insist, but it will be nice if this file and `tensor.hpp` are reformatted a bit to be more aligned with the typical mfem style -- e.g. align at 80 columns, don't add `*` on new lines of `/** ... */` comment, etc. (see e.g. `densemat.hpp`)